### DISCUSSION TITLE
Image preload/priority breakpoints

### DISCUSSION THREAD

**Discussion by topaxi** [vercel/next.js#43756] [Ideas]
### Describe the feature you'd like to request

The next image property `priority` allows preloading an image. Although an image might not be the LCP for every viewport it would be nice to be able to tell next image on which breakpoints it is a priority.

### Describe the solution you'd like

Instead of `priority` being a `boolean` prop, it would be nice if it would take a media query in addition to just a `boolean`.

Example:

```jsx
<Image fill src="/my/image.jpg" priority="(min-width: 1200px)" />
```

### Describe alternatives you've considered

We are currently preloading an LCP image that is not being shown on mobile, alternatively we could just disable the priority/preloading for Desktop altogether.

Another alternative we thought of was rendering a different page for mobile/desktop, but this seems to be quite complex and difficult.

**Comment by navin-moorthy** [comment]
Would be nice to have the `media` support in `ReactDOM.preload`

```patch
const preloadImgProps = getImgProps(imageProps);
ReactDOM.preload(preloadImgProps.props.src, {
	as: "image",
	imageSrcSet: preloadImgProps.props.srcSet,
+	media: "(min-width: 640px)",
	fetchPriority: "high",
});
```
